Resources/Public/JavaScript/paste-reference.js

```diff
@@ -0,0 +1,225 @@
+  const elementTitle = top.itemOnClipboardTitle != undefined ? top.itemOnClipboardTitle : "["+TYPO3.lang['tx_paste_reference_js.modal.labels.no_title']+"]";
+  const title = (TYPO3.lang['paste.modal.title.paste'] || 'Paste record') + ': "' + elementTitle + '"';
+  const severity = (typeof top.TYPO3.Severity[$element.data('severity')] !== 'undefined') ? top.TYPO3.Severity[$element.data('severity')] : top.TYPO3.Severity.info;
+  let buttons = [];
+  let content = '';
+
+  if ($element.hasClass('t3js-paste-copy')) {
+    content = TYPO3.lang['tx_paste_reference_js.modal.pastecopy'] || 'How do you want to paste that clipboard content here?';
+    buttons = [
+      {
+        text: TYPO3.lang['paste.modal.button.cancel'] || 'Cancel',
+        active: true,
+        btnClass: 'btn-default',
+        trigger: (evt, modal) => modal.hideModal(),
+      },
+      {
+        text: TYPO3.lang['tx_paste_reference_js.modal.button.pastecopy'] || 'Paste as copy',
+        btnClass: 'text-white btn-' + top.TYPO3.Severity.getCssClass(severity),
+        trigger: function (evt, modal) {
+          modal.hideModal();
+          DragDrop.default.onDrop(top.itemOnClipboardUid, $element, evt);
+        }
+      },
+      {
+        text: TYPO3.lang['tx_paste_reference_js.modal.button.pastereference'] || 'Paste as reference',
+        btnClass: 'text-white btn-' + top.TYPO3.Severity.getCssClass(severity),
+        trigger: function (evt, modal) {
+          modal.hideModal();
+          DragDrop.default.onDrop(top.itemOnClipboardUid, $element, evt, 'reference');
+        }
+      }
+    ];
+    if (top.pasteReferenceAllowed * 1 !== 1) {
+      buttons.pop();
+    }
+  } else {
+    content = TYPO3.lang['paste.modal.paste'] || 'Do you want to move the record to this position?';
+    buttons = [
+      {
+        text: TYPO3.lang['paste.modal.button.cancel'] || 'Cancel',
+        active: true,
+        btnClass: 'btn-default',
+        trigger: (evt, modal) => modal.hideModal(),
+      },
+      {
+        text: TYPO3.lang['paste.modal.button.paste'] || 'Move',
+        btnClass: 'btn-' + Severity.getCssClass(severity),
+        trigger: function (evt, modal) {
+          modal.hideModal();
+          DragDrop.default.onDrop(top.itemOnClipboardUid, $element, null);
+        }
+      }
+    ];
+  }
+  if (url !== null) {
+    const separator = (url.indexOf('?') > -1) ? '&' : '?';
+    const params = $.param({data: $element.data()});
+    Modal.loadUrl(title, severity, buttons, url + separator + params);
+  } else {
+    Modal.show(title, content, severity, buttons);
+  }
+};
+
+/**
+ * activates the paste into / paste after and fetch copy from another page icons outside of the context menus
+ */
+Paste.activatePasteIcons = function () {
+
+  $('.t3js-page-new-ce').each(function () { // t3-page-ce-wrapper-new-ce
+
+    if (!$(this).find('.icon-actions-plus').length) { // icon-actions-add
+      return true;
+    }
+
+    // @todo: check what does it do in detail and why?
```

Comment:
merged with that comment in place ? merging something not knowign what it does and why ?

Reply:
I didn't discover usage of the class `t3js-page-lang-column` at least down to TYPO3 version v9 and Paste.js of `backend` has the same code line.
So I suspect this code didn't have any impact for a long time. While the class can be found in some code in Versions 9, 10, 11, it is entirely removed in v12.

Reply:
fixed with 61883dc.
The JS is not required anymore like explained in the commit.